### PR TITLE
Don't confuse with sub-node

### LINKED ISSUE #25667: REST API bug when starting an container
**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:40:59 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 21:40:59 2016
 OS/Arch:      linux/amd64

```

**Output of `docker info`:**

```
Containers: 291
 Running: 9
 Paused: 0
 Stopped: 282
Images: 62
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 688
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options:
Kernel Version: 3.16.0-4-amd64
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.41 GiB
Name: tyr
ID: BOKP:BDNT:4NX7:D26W:3M7L:Y2TV:RSMA:EYGV:C2CS:REFP:EVNH:3EPY
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 56
 Goroutines: 77
 System Time: 2016-08-12T17:32:13.892406816+02:00
 EventsListeners: 1
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
Insecure Registries:
 127.0.0.0/8

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. Create a container but do not start it yet
2. Try to start it via the REST API
3. Will fail with `starting container with HostConfig was deprecated since v1.10 and removed in v1.12`

**Describe the results you received:**
The container did not start and I am recieving an 400 HTTP Code instead.
Debug Logs from dockerd: 

**Describe the results you expected:**
Starting the container should just work fine if the post payload is empty or an empty json, e.g. `{}`

**Additional information you deem important (e.g. issue happens only occasionally):**

> Aug 12 17:29:21 tyr dockerd[9899]: time="2016-08-12T17:29:21.278363023+02:00" level=debug msg="Calling POST /v1.24/containers/2e5b53aef3845502580e9ee8acdcd57f967d761015210c05dc37a208438e8250/start"
> Aug 12 17:29:21 tyr dockerd[9899]: time="2016-08-12T17:29:21.278425649+02:00" level=debug msg="form data: {}"
> Aug 12 17:29:21 tyr dockerd[9899]: time="2016-08-12T17:29:21.278449660+02:00" level=error msg="Handler for POST /v1.24/containers/2e5b53aef3845502580e9ee8acdcd57f967d761015210c05dc37a208438e8250/st$rt returned error: starting container with HostConfig was deprecated since v1.10 and removed in v1.12"
> 

There are also cases where everything just works fine:

> Aug 12 17:37:14 tyr dockerd[9899]: time="2016-08-12T17:37:14.769505957+02:00" level=debug msg="Calling POST /v1.24/containers/0af742407d8631299b5e37da4cfc10374d26294e6b4b5c7dce89eff1e8ec581a/start"
> Aug 12 17:37:14 tyr dockerd[9899]: time="2016-08-12T17:37:14.769582892+02:00" level=debug msg="form data: {}"
> Aug 12 17:37:14 tyr dockerd[9899]: time="2016-08-12T17:37:14.770618666+02:00" level=debug msg="container mounted via layerStore: /var/lib/docker/aufs/mnt/f01da2f0f153a1a0f80b4e28fabd86dd23b3af3d694c9af92aee133793b0ba5e"

Is there a way to log eveything that comes in via the API user headers etc? Because I can't see a difference between those two POST requests

### PR DESCRIPTION
**Pull request by KostyaSha**
closes #25667

### CODE REVIEW

File: api/server/router/container/container_routes.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (s *containerRouter) postContainersStart(ctx context.Context, w http.Respon
 	// A non-nil json object is at least 7 characters.
 	if r.ContentLength > 7 || r.ContentLength == -1 {
 		if versions.GreaterThanOrEqualTo(version, "1.24") {
-			return validationError{fmt.Errorf("starting container with HostConfig was deprecated since v1.10 and removed in v1.12")}
+			return validationError{fmt.Errorf("starting container with any parameters was deprecated since v1.10 and removed in v1.12")}
```
Comment (thaJeztah) [inline api/server/router/container/container_routes.go]: Historically, it only accepted a HostConfig as body; https://docs.docker.com/v1.4/reference/api/docker_remote_api_v1.11/#start-a-container, and always referenced it as `HostConfig` in the deprecation message; https://docs.docker.com/engine/reference/api/docker_remote_api_v1.18/#/start-a-container

Perhaps we should change it to be clearer that this is about the *body* (i.e. not query parameters)

Comment (KostyaSha) [inline api/server/router/container/container_routes.go]: Ah, right here `HostConfig` was used as is, but in create it was sub-node.
Will change to body.

**Comment by KostyaSha** [comment]
Hm.. but commit is signed even in GH UI.

**Comment by tonistiigi** [comment]
@KostyaSha Your second commit is not. Also please squash.

**Comment by KostyaSha** [comment]
@tonistiigi 
screenshot 2016-08-17 01 51 30

https://help.github.com/articles/about-pull-request-merge-squashing/

**Comment by tonistiigi** [comment]
@KostyaSha We check for the `Signed-off-by` in each commit message. GPG signatures in git are different, we don't require those. https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

**Comment by thaJeztah** [comment]
I usually just do both; `git commit -s -S`

**Comment by KostyaSha** [comment]
should be fixed now

**Comment by tonistiigi** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM, thanks!

**Pull request merged**